velox/common/base/ConcurrentCounter.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+#include <new>
+
+#include "velox/common/base/BitUtil.h"
+#include "velox/common/base/Exceptions.h"
+
+namespace facebook::velox {
+
+/// The class provides concurrent updates to a counter with minimum lock
+/// contention. The counter is N-way sharded internally. Each update goes to one
+/// of the sharded counter based on the update thread id.
+template <class T>
+class ConcurrentCounter {
+ public:
+  ConcurrentCounter(size_t numShards, T initValue)
```

Comment:
Would you document this constructor?

Why do we need initValue? Can we not use T()?

Reply:
Good point!